### ISSUE TITLE
Seeking clarification re: changes to accessible description when a form field has focus

### ISSUE THREAD

**Issue by jscholes** [w3c/aria#1200]
#### Background

I am seeking some clarification surrounding how user agents and screen readers should react in the following scenarios:

1. `aria-describedby` is applied to a `<input>`, its value referencing an element containing text, while the `<input>` still has focus.
2. The text content of the container pointed to by `aria-describedby` on a `<input>` is updated while the `<input>` still has focus.

This is to support a use case whereby field validation occurs as a user is typing, and an error message or positive confirmation of validation is added to the DOM after the field.  That message should be programmatically associated with the field as its accessible description, but should also be automatically announced by a screen reader.

My initial thought was to link the message to the field using `aria-describedby`, but also speak the text via a live region.  However, with certain browser and screen reader combinations, this results in the message being announced twice, if the field has focus when validation results come in.

#### Test case

When the below URL is navigated to, the "Username" field will receive focus automatically.  At page load time, the field has no `aria-describedby` attribute.  After 4000ms, `aria-describedby` is added to the field, referencing an element containing an initial help message.  After 9000ms, the content of that message is updated.  No live region is present.

https://jscholes.github.io/aria-describedby_with_focus.html

#### Expected results

A screen reader will stay silent following the initial focus event.  That is, the changes to the field's accessible description should not be announced, unless the user explicitly navigates away from the field and back in or uses some other mechanism to read it.

##### Chrome with JAWS and NVDA

The screen reader speaks when:

* `aria-describedby` is added to the field while it has focus; and
* the content of the element pointed to by `aria-describedby` is modified while the field has focus.

##### Firefox with both JAWS and NVDA

The screen reader will speak when `aria-describedby` is initially added to the field while it has focus, but not if the content of the element pointed to by `aria-describedby` is modified (again while the field has focus).

##### iOS Safari with VoiceOver

No changes to accessible description are announced, with the keyboard hidden or shown.

CC @sinabahram

**Comment by JAWS-test** [comment]
It would probably have to be specified in the https://w3c.github.io/accname/ whether changes to the name and descriptions would have to be output when the element has the focus. However, I suspect that this is not the case. Such changes can only be communicated via Live Regions.

**Comment by aleventhal** [comment]
Looks like Chrome does fire description change events, at least for UIA, IA2 and ATK.

The IA2 event is fired here:
https://cs.chromium.org/chromium/src/content/browser/accessibility/browser_accessibility_com_win.cc?type=cs&q=event_object_descriptionchange+file:%5C.cc&sq=package:chromium&g=0&l=1543

We have expectations files for ia2, uia and atk here:
https://cs.chromium.org/search/?q=file:description-change-expected&sq=package:chromium&type=cs

**Comment by jnurthen** [comment]
@aleventhal was that done in response to a spec (and we can file bugs against other UAs) or was it not? 
If it wasn't in repsonse to a spec where would you like to see it specified? I'm presuming core-aam?

**Comment by scottaohara** [comment]
There’s the change event already specified in core aam

https://www.w3.org/TR/core-aam-1.2/#event-aria-describedby
